src/collector.js

```diff
@@ -22,6 +24,12 @@ export default class Collector {
   }
 
   _addMessage(type, opts, _Message=Message) {
+    if (this.config.scanFile && opts.file) {
+      if (!this.config.scanFile.some(v => v === opts.file)) {
+        return;
+      }
+    }
```

Comment:
I might be missing some context but I don't get this code, can we add a comment to why we skip here?